foundry.toml

```diff
@@ -1,13 +1,58 @@
+# Full reference https://github.com/foundry-rs/foundry/tree/master/config
+
 [profile.default]
-src = "src"
-out = "out"
-libs = ["lib"]
-verbosity = 3
-solc_version = '0.8.20'
+  auto_detect_solc = false
+  block_number = 19537700 # Mar-29-2024 05:52:47 AM +UTC
+  bytecode_hash = "none"
+  cbor_metadata = false
+  fuzz = { runs = 1_000 }
+  gas_reports = ["*"]
+  libs = ["lib"]
+  optimizer = true
+  optimizer_runs = 10_000
+  out = "out"
+  script = "script"
+  solc = "0.8.20"
+  src = "src"
+  test = "test"
+  evm_version = 'london'
```

Comment:
make evm_version "paris" if we're targeting evms or "dencun" if we're targeting ethereum